.github/workflows/benchmarks.yml

```diff
@@ -143,6 +125,83 @@ jobs:
         with:
           path: asv-compare-changed-output.log
 
+      - name: Run benchmarks for base to head commits of PR
+        if: github.event_name == 'pull_request_target'
+        run: |
+          echo $(git rev-parse HEAD) > commit_hashes.txt
+          echo $(git merge-base HEAD upstream/master) >> commit_hashes.txt
+
+          asv run HASHFILE:commit_hashes.txt | tee asv-output-PR.log  
+            if grep -q failed asv-output-PR.log; then 
+              echo "Some benchmarks have failed!"
+              exit 1
+            fi
+
+      - name: Generate Graphs and HTML of PR
+        if: github.event_name == 'pull_request_target'
+        run: |
+          asv publish
+
+      - name: Delete env files of PR
+        if: github.event_name == 'pull_request_target'
+        run: rm -r .asv/env
+
+      - name: Set destination directory
+        if: github.event_name == 'pull_request_target'
+        run: |
+          BRANCH=$(echo ${GITHUB_REF#refs/heads/})
+          if [[ $EVENT == push ]] || [[ $EVENT == workflow_dispatch ]]; then
+
+            if [[ $BRANCH == $DEFAULT ]]; then
+              echo "DEST_DIR=" >> $GITHUB_ENV
+            else
+              echo "DEST_DIR=branch/$BRANCH" >> $GITHUB_ENV
+            fi
+
+          elif [[ $EVENT == pull_request_target ]]; then
+            echo "DEST_DIR=pull/$PR" >> $GITHUB_ENV
+
+          else
+            echo "Unexpected event trigger $EVENT"
+            exit 1
+
+          fi
+
+          cat $GITHUB_ENV
+        env:
+          DEFAULT: ${{ github.event.repository.default_branch }}
+          EVENT: ${{ github.event_name }}
+          PR: ${{ github.event.number }}
+
+      - name: Set clean branch option
+        if: github.event_name == 'pull_request_target'
+        run: |
+          if [[ $EVENT == workflow_dispatch ]]; then
+            echo "CLEAN_BRANCH=true" >> $GITHUB_ENV
+
+          else
+            echo "CLEAN_BRANCH=false" >> $GITHUB_ENV
+
+          fi
+
+          cat $GITHUB_ENV
+        env:
+          EVENT: ${{ github.event_name }}
+
+      - name: Deploy ${{ env.DEST_DIR }}
+        if: github.event_name == 'pull_request_target'
+        uses: peaceiris/actions-gh-pages@v4
+        with:
+          personal_token: ${{ secrets.BOT_TOKEN }}
+          publish_branch: ${{ env.DEPLOY_BRANCH }}
+          publish_dir: .asv/html
```

Comment:
Perhaps we can remove this? Since .asv/html folder no longer exists?

Reply:
Or this is option mandatory?

Reply:
Yes, for deploying

Reply:
I see